### PR TITLE
Change async to run_async for Python 3.7

### PR DESCRIPTION
**Pull request by alexboily-ls**
`async` is now a [reserved keyword in Python 3.7](https://docs.python.org/3/whatsnew/3.7.html#changes-in-python-behavior), causing syntax errors in `hiveserver2.py`.

The syntax errors are restricted to argument names, which were changed from `async` to `run_async`. Camel-cased `runAsync` on line 1032 was kept to avoid issues with Thrift generated code.

### CODE REVIEW

**Comment by cpcloud** [comment]
Can someone merge this patch? This is blocking downstream support of Python 3.7

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ply==3.11
```
Comment (cpcloud) [inline requirements.txt]: Pinning exact versions has the potential to cause a lot of headaches for downstream users of this library, especially across major Python versions.

Why did you add a requirements file?

File: impala/hiveserver2.py [hunk 2]
```diff
@@ -1024,11 +1024,11 @@ def close(self):
         req = TCloseSessionReq(sessionHandle=self.handle)
         self._rpc('CloseSession', req)
 
-    def execute(self, statement, configuration=None, async=False):
+    def execute(self, statement, configuration=None, run_async=False):
```
Comment (cpcloud) [inline impala/hiveserver2.py]: If you really want to be thorough here you can maintain backwards compatibility by going through a deprecation cycle.

You would do this by setting the default value of `run_async` to `None`, adding `**kwargs` to the method signature, checking if `run_async` is not `None` and using that if so else use `kwargs.get('async', False)`. If you do decide to do this then you should also check if `'async' in kwargs` and give a warning to users that their code needs to be updated to use `run_async` because in the next release you will not accept this keyword argument.

**Comment by wesm** [comment]
I can help with this -- @cpcloud are you a maintainer on this package?

**Comment by cpcloud** [comment]
@wesm I am not

**Comment by wesm** [comment]
@gregrahn can someone from Cloudera add @cpcloud as an admin on this project? It isn't being maintained very much, but we can at least merge incoming PRs and cut releases to PyPI

**Comment by gregrahn** [comment]
Added @cpcloud -- ping me with any issues.

**Comment by cpcloud** [comment]
@wesm @gregrahn Thanks guys. I'm cranking out a fix for the Python 3.7 issue. Can you give me access to PyPI as well? My username is Charles.Cloud

File: impala/hiveserver2.py (hunk 2 above)
Comment (cpcloud) [reply impala/hiveserver2.py]: @wesm Actually, are you aware of whether this method `HS2Session.execute` is part of the public API? It looks like it might not be in which case we don't have to handle backwards compatibility here.

Comment (cpcloud) [reply impala/hiveserver2.py]: After digging around a bit I don't think this method is part of the public API. Users go through `connect -> HiveServer2Connection -> .cursor() -> HiveServer2Cursor -> execute/execute_async` and never see the `HS2Session.execute` calls.

**Comment by cpcloud** [comment]
Closing in favor of #322

**Comment by cpcloud** [comment]
@alexboily-ls I preserved your commits in #322.

**Comment by wesm** [comment]
I think it's OK to ignore backwards compatibility issues here
